lib/AST/ASTContext.cpp

```diff
@@ -908,6 +914,75 @@ FuncDecl *ASTContext::getEqualIntDecl() const {
   return nullptr;
 }
 
+FuncDecl *ASTContext::getMutatingXorIntDecl() const {
+  if (Impl.MutatingXorIntDecl)
+    return Impl.MutatingXorIntDecl;
+
+  auto intType = getIntDecl()->getDeclaredType();
+  auto inoutIntType = InOutType::get(intType);
+  SmallVector<ValueDecl *, 30> xorFuncs;
```

Comment:
Nit: Are there actually 30 of these?  Usually SmallVector sizes are powers of 2, so 32 would be more apropos